pkg/build/api/types.go

```diff
@@ -616,6 +616,9 @@ type JenkinsPipelineBuildStrategy struct {
 
 	// Jenkinsfile defines the optional raw contents of a Jenkinsfile which defines a Jenkins pipeline build.
 	Jenkinsfile string
+
+	// Env contains additional environment variables you want to pass into a build pipeline
+	Env []kapi.EnvVar
```

Comment:
The problem with validation would be breaking anyone who already set it.  But adding validation when we create a field is ok.  I'll give it a pass for now.